crates/packaging/src/https.rs

```diff
@@ -275,9 +275,17 @@ pub fn download_and_hash(
         Encoding::new(content_encoding, url)?
     };
 
-    // Use .take to prevent a malicious server from sending back bytes
-    // until system resources are exhausted!
-    decompress_into(dest_dir, encoding, resp.take(max_download_bytes))
+    if let Some(content_len) = resp.content_length() {
+        // Print download progress to stdout if we know the content length
+        //
+        // Use .take to prevent a malicious server from sending back bytes
+        // until system resources are exhausted!
+        let resp = ProgressReporter::new(resp.take(max_download_bytes), content_len as usize);
+
+        decompress_into(dest_dir, encoding, resp)
+    } else {
+        decompress_into(dest_dir, encoding, resp.take(max_download_bytes))
+    }
```

Comment:
Just pulling this to the top so that it is always applied even if someone adds a new branch in the future. I think this should still work correctly, but haven't tested. Might have to pull the content length option into a variable for this to work correctly depending on how the response type works.
```suggestion
    // Use .take to prevent a malicious server from sending back bytes
    // until system resources are exhausted!
    let resp = resp.take(max_download_bytes);

    if let Some(content_len) = resp.content_length() {
        // Print download progress to stdout if we know the content length
        let resp = ProgressReporter::new(resp, content_len as usize);

        decompress_into(dest_dir, encoding, resp)
    } else {
        decompress_into(dest_dir, encoding, resp)
    }
```